tests/integration-tests/tests/storage/snapshots_factory.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+import threading
+import time
+from collections import namedtuple
+
+import boto3
+from retrying import retry
+
+from fabric import Connection
+
+SnapshotConfig = namedtuple("ClusterConfig", "ssh_key key_name vpc_id master_subnet_id")
+
+
+class SnapshotsFactory:
+    """Manage creation and destruction of volume snapshots."""
+
+    master_snapshot = None
+    master_snapshot_region = None
+    factory_lock = threading.Lock()
+    snapshots = set()
+
+    def __init__(self):
+        self.config = None
+        self.instance = None
+        self.volume = None
+        self.snapshot = None
+        self.security_group_id = None
+
+    def get_snapshot(self, request, subnet_id, region):
+        """
+        Create a snapshot in a given region.
+        :param request: The current request
+        :param subnet_id: The subnet id where to get the snapshot
+        :param region: The region where to get the snapshot
+        """
+        # If a master snapshot already exists just a copy is done
+        with self.__class__.factory_lock:
```

Comment:
ok I think I get why we have this static class variables and a locking mechanism. In general I'd rather keep the testing code as simple as possible (we don't want to write tests for the tests :))

What if we keep this code as a standalone script to create a snapshot in all regions and we mark these snapshots are public so we do not add any overhead to the tests execution and by marking it as public everybody can run the tests with that snapshot. Let's talk about this.